packages/open-next/src/adapters/revalidate.ts

```diff
@@ -0,0 +1,57 @@
+import { SQSEvent } from "aws-lambda";
+import fs from "node:fs";
+import path from "node:path";
+import { request } from "node:https";
+import { IncomingMessage } from "node:http";
+
+interface PrerenderManifest {
+  preview: {
+    previewModeId: string;
+    previewModeSigningKey: string;
+    previewModeEncryptionKey: string;
+  };
+}
+
+function loadPrerenderManifest() {
+  const filePath = path.join("prerender-manifest.json");
+  const json = fs.readFileSync(filePath, "utf-8");
+  return JSON.parse(json) as PrerenderManifest;
+}
+
+export const handler = async (event: SQSEvent) => {
+  const { preview } = loadPrerenderManifest();
+  for (const record of event.Records) {
+    const body = record.body;
+    const data = JSON.parse(body);
+    const url = data.url;
+    console.log(`Revalidating stale page ${url}`);
+
+    try {
+      // We fire off a HEAD request to the page to revalidate it
+      // This will trigger the page to be re-rendered and cached in S3
+      // We use the previewModeId to ensure the page is revalidated in a blocking way in lambda
+      // https://github.com/vercel/next.js/blob/1088b3f682cbe411be2d1edc502f8a090e36dee4/packages/next/src/server/api-utils/node.ts#L353
+      await new Promise<IncomingMessage>((resolve, reject) => {
+        const req = request(
+          `https://${process.env.HOST}${url}`,
+          {
+            method: "HEAD",
+            headers: { "x-prerender-revalidate": preview.previewModeId },
+          },
+          (res) => {
+            resolve(res);
+          }
+        );
+        req.on("error", (err) => reject(err));
+        req.end();
+      });
+      console.log(`Revalidated stale page ${url}`);
+    } catch (e) {
+      console.error("Failed to revalidate stale page", url);
+      console.error(e);
+    }
+  }
+  return {
+    statusCode: 200,
+  };
```

Comment:
Yes you're right, this one actually give me some headache due to the 5 min deduplication window

Reply:
On the dupe window, might be worth looking at how the original SQS handled this... (serverless-nextjs)

Reply:
After struggling a bit that's what i did : 
https://github.com/serverless-nextjs/serverless-next.js/blob/e6367b585fb98608cd2e9327e2c8d4058ba73b00/packages/libs/lambda-at-edge/src/lib/triggerStaticRegeneration.ts#L53
They also have a lastModified here, but we don't have access to this if we don't go through the interceptor

Reply:
Thinking out loud:
Mini-interceptor-like-middleware that sticks `lastModified` on the `req` somewhere.  Picks it up on the other side where `sendToRevalidateSqs` is called.

Reply:
That should work, but should we go this route ?
Doing something like that would add something like 50-150 ms on every SSG or ISR request since we need to check S3
The issue only occurs when the revalidation is less than 5 minutes, and there is a pretty easy fix for the user. 
For example returning a `Date.now()` from getStaticProps and using this data in a hidden div on the page. This would ensure that the `etag` is different
Maybe just adding something on the readme to explain this ?